### PR TITLE
refactor: completion

### PR DESCRIPTION
**Pull request by xiaoyijun**
## Summary

* refactor: completion
* add bound to generic type
### Todo
* LOG-537: Exceptions

## Linear Issue Reference

* LOG-1489

## Testing

UT & Test in App
image

### CODE REVIEW

File: android-sample/app/src/main/java/io/logto/demo/fragment/LoginFragment.kt [hunk 1]
```diff
@@ -37,9 +37,8 @@ class LoginFragment : Fragment() {
             logtoClient.signInWithBrowser(
                 requireActivity(),
                 "io.logto.android://io.logto.sample/callback",
-            ) { throwable: Throwable?, result: String? ->
-                println(throwable)
-                println(result)
+            ) { throwable: Throwable? ->
+                throwable?.let { println("Sign In Failed: ${it.printStackTrace()}") } ?: println("Sign In Success")
```
Comment (demonzoo) [inline android-sample/app/src/main/java/io/logto/demo/fragment/LoginFragment.kt:41]: nit: The opposite of `failed` should be `successful`

File: android-sdk/android/src/main/kotlin/io/logto/sdk/android/LogtoClient.kt [hunk 2]
```diff
@@ -72,7 +71,7 @@ open class LogtoClient(
                 refreshToken = codeTokenResponse.refreshToken
                 idToken = codeTokenResponse.idToken
 
-                completion.onComplete(null, codeTokenResponse.accessToken)
```
Comment (demonzoo) [inline android-sdk/android/src/main/kotlin/io/logto/sdk/android/LogtoClient.kt:75]: Is the access token not needed at all in the completion logic?

Comment (xiaoyijun) [reply android-sdk/android/src/main/kotlin/io/logto/sdk/android/LogtoClient.kt:75]: We get access tokens by `getAccessToken`.
Only results are needed here

**Review by wangsijie — APPROVED** [review]

File: android-sample/app/src/main/java/io/logto/demo/fragment/LoginFragment.kt (hunk 1 above)
Comment (xiaoyijun) [reply android-sample/app/src/main/java/io/logto/demo/fragment/LoginFragment.kt]: updated

**Pull request merged**